src/main/java/tds/exam/results/services/impl/ExamResultsServiceImpl.java

```
@@ -0,0 +1,80 @@
+    private final AssessmentService assessmentService;
+    private final JAXBContext contextObj;
+
+    @Autowired
+    public ExamResultsServiceImpl(final ExamService examService,
+                                  final SessionService sessionService,
+                                  final AssessmentService assessmentService) throws JAXBException {
+        this.examService = examService;
+        this.sessionService = sessionService;
+        this.assessmentService = assessmentService;
+
+        contextObj = JAXBContext.newInstance(TDSReport.class);
+        Marshaller marshallerObj = contextObj.createMarshaller();
+        marshallerObj.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+    }
+
+    @Override
+    public TDSReport findExamResults(final UUID examId) throws SAXException, IOException, JAXBException {
+        TDSReport report = new TDSReport();
+        final ExpandableExam expandableExam = examService.findExpandableExam(examId);
+        final Exam exam = expandableExam.getExam();
+        final Session session = sessionService.findSessionById(exam.getSessionId());
+        final Assessment assessment = assessmentService.findAssessment(exam.getClientName(), exam.getAssessmentKey());
+
+        report.setOpportunity(OpportunityMapper.mapOpportunity(expandableExam, session, assessment));
+        report.setExaminee(ExamineeMapper.mapExaminee(expandableExam));
+        report.setTest(TestMapper.mapTest(assessment));
+        CommentMapper.mapComments(report.getComment(), expandableExam);
+
+        validateReport(report);
+
+        return report;
+    }
+
+    private void validateReport(final TDSReport results) throws JAXBException, SAXException, IOException {
+        JAXBSource source = new JAXBSource(contextObj, results);
+        SchemaFactory sf = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema schema = sf.newSchema(new File("src/main/xsd/TestResultsTransmissionFormat_Schema.xsd"));
```

Rather than get to the schema this way what if we did something like this:

```
try(InputStream is = Thread.currentThread().getContextClassLoader().getResourceAsStream("xsd/TestResultsTransmissionFormat_Schema.xsd")) {
  Schema schema = sf.newSchema(new StreamSource(is));
  Validator validator = schema.newValidator();
  //validate
}
```

Also, is this something that could be created at startup (or injected) rather than each time we want to validate?

---

If we separated the validator into a separate class that is injected we could also catch all these exceptions and consolidate them into one RuntimeException?  Since we aren't actually going to do anything with the exception and just let it bubble.